pkg/services/sso/redhatsso_service.go

```diff
@@ -0,0 +1,136 @@
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorFailedToCreateServiceAccount, err, "failed to create service account")
+	}
+
+	return convertServiceAccountDataToAPIServiceAccount(&serviceAccount), nil
+}
+
+func (r *redhatssoService) DeleteServiceAccount(accessToken string, ctx context.Context, clientId string) *errors.ServiceError {
+	err := r.client.DeleteServiceAccount(accessToken, clientId)
+	if err != nil { //5xx
+		return errors.NewWithCause(errors.ErrorFailedToDeleteServiceAccount, err, "failed to delete service account")
+	}
+	glog.V(5).Infof("deleted service account clientId = %s ", clientId)
+	return nil
+}
+
+func (r *redhatssoService) ResetServiceAccountCredentials(accessToken string, ctx context.Context, clientId string) (*api.ServiceAccount, *errors.ServiceError) {
+	serviceAccount, err := r.client.RegenerateClientSecret(accessToken, clientId)
+	if err != nil { //5xx
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to reset service account credentials")
+	}
+	return convertServiceAccountDataToAPIServiceAccount(&serviceAccount), nil
+}
+
+func (r *redhatssoService) ListServiceAcc(accessToken string, ctx context.Context, first int, max int) ([]api.ServiceAccount, *errors.ServiceError) {
+	serviceAccounts, err := r.client.GetServiceAccounts(accessToken, first, max)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to collect service accounts")
+	}
+
+	var saList []api.ServiceAccount
+	for _, sa := range serviceAccounts {
+		saList = append(saList, *convertServiceAccountDataToAPIServiceAccount(&sa))
+	}
+	return saList, nil
+}
+func (r *redhatssoService) RegisterKasFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) (*api.ServiceAccount, *errors.ServiceError) {
+	return nil, nil
+}
+func (r *redhatssoService) DeRegisterKasFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) *errors.ServiceError {
+	return nil
+}
+func (r *redhatssoService) GetServiceAccountById(accessToken string, ctx context.Context, id string) (*api.ServiceAccount, *errors.ServiceError) {
+	return r.GetServiceAccountByClientId(accessToken, ctx, id)
+}
+func (r *redhatssoService) GetServiceAccountByClientId(accessToken string, ctx context.Context, clientId string) (*api.ServiceAccount, *errors.ServiceError) {
+	serviceAccount, err := r.client.GetServiceAccount(accessToken, clientId)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorServiceAccountNotFound, err, "service account not found clientId %s", clientId)
+	}
+	return convertServiceAccountDataToAPIServiceAccount(&serviceAccount), nil
+}
+func (r *redhatssoService) RegisterConnectorFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) (*api.ServiceAccount, *errors.ServiceError) {
+	return nil, nil
+}
+func (r *redhatssoService) DeRegisterConnectorFleetshardOperatorServiceAccount(accessToken string, agentClusterId string) *errors.ServiceError {
+	return nil
```

Comment:
@akoserwal It is not clear to me how we are going to implement this method with the new api

Reply:
``` 
    accessToken, err := r.client.GetToken()
	if err != nil {
		return errors.NewWithCause(errors.ErrorGeneral, err, "failed to deregister service account")
	}
         err := r.client.DeleteServiceAccount(accessToken, agentClusterId)
	if err != nil {
		return errors.NewWithCause(errors.ErrorFailedToDeleteServiceAccount, err, "Failed to delete service account: %s", agentClusterId)
	}
	return nil
```

Reply:
ok, so `agentClusterId` is the same as `clientId`.

Reply:
We need to check `DeRegisterConnectorFleetshardOperatorServiceAccount` is called should always pass the client id stored in the database. We can tackle that part in separate jira.